kubemarine/cri/containerd.py

```diff
@@ -24,6 +24,17 @@
 from kubemarine.core import utils
 from kubemarine.core.group import NodeGroup, RunnersGroupResult, CollectorCallback
 
+def add_yum_repo_docker(group: NodeGroup)-> RunnersGroupResult:
```

Comment:
Is specifying repo and extra package in cluster.yaml not working ?
https://github.com/Netcracker/KubeMarine/blob/main/examples/cluster.yaml/allinone-cluster-ol7.yaml#L33
https://github.com/Netcracker/KubeMarine/blob/main/examples/cluster.yaml/allinone-cluster-ol7.yaml#L56

Reply:
There are several reasons why I did it this way. Let me tell you about the most important ones.
Firstly, the docker and containerd packages are required to install kubernetes on any OS. In Debian-like distributions, these packages are available immediately, while in systems based on RHEL, it is necessary to connect an additional repository. Therefore, it is logical to transfer the mandatory step when deploying a kubernetes cluster to flow kubemarine. This will make it easier for users to configure the deployment. Isn't that the purpose of any automation? In addition, there is the podman. In ubuntu 20.04, for example, the repository with it needs to be connected separately. I or anyone else in the future will be able to add this function to the cri.add_repo.
Secondly, it is unsafe to connect repositories without GPG authentication. And if you enable authentication in the configuration file that you propose as a template, you will always get an error in installation packages, because the add_repo function only adds the repository to the predefined.repo file without GPG-key. 
Therefore, I consider it justified to introduce a separate step to add the required third-party repositories.

Reply:
Hello @krang404,
To clarify, the main goal of the application is to install Kubernetes cluster in production environment. Thus, the main case is when public internet resources are unavailable. The following solutions are developed for this: [Installation without Internet Resources](https://github.com/Netcracker/KubeMarine/blob/main/documentation/Installation.md#installation-without-internet-resource), [package_manager](https://github.com/Netcracker/KubeMarine/blob/main/documentation/Installation.md#package_manager). They allow to specify other custom sources of packages, images, not only the missed docker repo.

This PR is about installation in public environment. It is also supported but should not contradict to the private mode. You add public repo in any mode that looks at least strange.

Returning to your comment. In other words, the approach suggested by @koryaga works in unsafe mode, doesn't it? When repositories are specified in `cluster.yaml`, the installation is automated, and the same approach is applicable for ubuntu 20.04. The flaw related to authentication can potentially be fixed by extending of the [PackageManager.add_repo()](https://github.com/Netcracker/KubeMarine/blob/0.19.0/kubemarine/packages.py#L512) procedure (and corresponding installation task `prepare.package_manager.configure`) in generic way for both public and private environments. What we can think of is again possible extending of the `prepare.package_manager.configure` task to install some repositories by default in some (to be developed) cases.

Reply:
Hello @ilia1243 First of all, thank you for noticing that the lack of access to the docker repository is not handled in any way. I fixed it. But I also want to tell you that in an environment where there is no SNAT on the Internet, there will also be no access to the repositories from the example above and this will also require a configuration of private resources to install. In such circumstances, it is enough for the user to do ``--exclude prepare.cri.add_repo``. Secondly, I think another way to add the necessary repositories corresponds to the Linux approach, where everything can be done in several ways. And while the alternative you're talking about is still in the thought experiment phase, this fix will help people who install a kubernetes cluster on nodes with RHEL-like operating systems.

Reply:
> lack of access to the docker repository is not handled in any way.

It is handled through `services.packages.package_manager.repositories` section of the configuration though without GPG-key. Please confirm that the installation works in this case.

> it is enough for the user to do `--exclude prepare.cri.add_repo`

Unfortunately, we cannot require from people to do `--exclude prepare.cri.add_repo` while they do not need to do that now. It can be considered as a some sort of breakage.

> And while the alternative you're talking about is still in the thought experiment phase, this fix will help people who install a kubernetes cluster on nodes with RHEL-like operating systems.

While you fix can work, we cannot include it if is contradicts to the design of the application. By design, installation of repositories happens through the dedicated _task_ and dedicated section of the configuration file. Support of GP code and default public repos may be developed in these task and section for arbitrary repositories in the future. After this is done, it will likely contradict to how you suggest to do that for the particular repository.

Lets' focus on support of RHEL9 family within this PR. This is really valuable for us. The two new features related to the repositories will be handles in the separate issue.

Reply:
OK. I really understand why you are focused on deploying in secure environments. And I respect your approach to the architecture of the application. So, I have removed adding the docker-ce repository from the installation process.

Reply:
#481